extensions/wikia/CommunityHeader/tests/NavigationTest.php

```diff
@@ -0,0 +1,391 @@
+<?php
+
+use \Wikia\CommunityHeader\Label;
+use \Wikia\CommunityHeader\Link;
+use \Wikia\CommunityHeader\Navigation;
+use \Wikia\Util\GlobalStateWrapper;
+
+class NavigationTest extends WikiaBaseTest {
+	/**
+	 * $this->setUp() is called only after data providers are checked
+	 * We need the classes ready before that
+	 */
+	private function manualSetUp() {
+		global $IP, $wgAutoloadClasses;
+		require_once( "$IP/extensions/wikia/CommunityHeader/CommunityHeader.setup.php" );
+	}
+
+	/**
+	 * @dataProvider exploreItemsProvider
+	 *
+	 * @param $globals
+	 * @param $expectedExploreItems
+	 */
+	public function testExploreItems( $globals, $expectedExploreItems ) {
+		$globals = new GlobalStateWrapper( [
+			'wgEnableCommunityPageExt' => $globals['wgEnableCommunityPageExt'],
+			'wgEnableDiscussions' => $globals['wgEnableDiscussions'],
+			'wgEnableForumExt' => $globals['wgEnableForumExt'],
+		] );
+
+		$result = $globals->wrap( function () {
+			return new Navigation();
+		} );
+
+		// used `array_values` to reset keys of array
+		$this->assertEquals( $expectedExploreItems, array_values( $result->exploreItems ) );
+	}
+
+	/**
+	 * @dataProvider discussLinkProvider
+	 *
+	 * @param $globals
+	 * @param $expected
+	 */
+	public function testDiscussLink( $globals, $expected ) {
+		$globals = new GlobalStateWrapper( [
+			'wgEnableDiscussions' => $globals['wgEnableDiscussions'],
+			'wgEnableForumExt' => $globals['wgEnableForumExt'],
+		] );
+
+		$result = $globals->wrap( function () {
+			return new Navigation();
+		} );
+
+		$this->assertEquals( $expected, $result->discussLink );
+	}
+
+	private function prepareExploreItems( $raw ) {
+		return array_map( function ( $rawItem ) {
+			return new Link(
+				new Label( $rawItem['label']['key'], $rawItem['label']['type'] ),
+				$rawItem['href'],
+				$rawItem['tracking']
+			);
+		}, $raw );
+	}
+
+	public function exploreItemsProvider() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function "exploreItemsProvider" has 259 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)